### PR TITLE
[RNMobile] Upgrade `compileSdkVersion` to 34 (Android)

### LINKED ISSUE #6142: Update `compileSdkVersion` to 34 in Gutenberg Mobile

### PR DESCRIPTION
**Pull request by fluiddot**
Related PRs:
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/6196
* https://github.com/wordpress-mobile/WordPress-Android/pull/19182

## What?

Upgrades the `compileSdkVersion` to 34 as part of the support for Android 14.

## Why?

Closes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6142.

## How?

Updates the value in all references of `compileSdkVersion`:
- `react-native-aztec`
- `react-native-bridge`
- `react-native-editor`

Bump `AztecEditor-Android` version to `1.8.0`, which also uses version 34 of `compileSdkVersion`.

**⚠️ NOTE:** In past PRs where we upgraded Android ([example](https://github.com/WordPress/gutenberg/pull/50731)), we also updated the React Native third-party libraries:
- Libraries published via [react-native-libraries-publisher](https://github.com/wordpress-mobile/react-native-libraries-publisher/blob/2746ac55f78dd540efb2fa8089f1e05a2f7899a0/package.json#L3-L12)
- [react-native-slider](https://github.com/wordpress-mobile/react-native-slider)
- [react-native-video](https://github.com/wordpress-mobile/react-native-video)

I assume that since we are only updating the `compileSdkVersion` and not `targetSdkVersion`, it's safe to leave them with an older version. But once we fully upgrade to Android 14, we'd need to update them.

## Testing Instructions

* Observe that all CI jobs pass.
* Smoke test the editor on Android.

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 
N/A

### CODE REVIEW

**Review by thomashorta — APPROVED** [review]
The changes look good to me. Thanks for helping with this! 🙇🏼

File: packages/react-native-aztec/android/build.gradle [hunk 1]
```diff
@@ -11,7 +11,7 @@ buildscript {
         espressoVersion = '3.0.1'
 
         // libs
-        aztecVersion = 'v1.6.4'
+        aztecVersion = 'v1.8.0'
```
Comment (fluiddot) [inline packages/react-native-aztec/android/build.gradle:14]: The Aztec library has been updated to incorporate https://github.com/wordpress-mobile/AztecEditor-Android/pull/1059.

**Comment by fluiddot** [comment]
> **⚠️ NOTE:** In past PRs where we upgraded Android ([example](https://github.com/WordPress/gutenberg/pull/50731)), we also updated the React Native third-party libraries:
> 
> * Libraries published via [react-native-libraries-publisher](https://github.com/wordpress-mobile/react-native-libraries-publisher/blob/2746ac55f78dd540efb2fa8089f1e05a2f7899a0/package.json#L3-L12)
> * [react-native-slider](https://github.com/wordpress-mobile/react-native-slider)
> * [react-native-video](https://github.com/wordpress-mobile/react-native-video)
> 
> I assume that since we are only updating the `compileSdkVersion` and not `targetSdkVersion`, it's safe to leave them with an older version. But once we fully upgrade to Android 14, we'd need to update them.

cc @thomashorta @ParaskP7 in case you have concerns/feedback about following the above approach.

File: packages/react-native-aztec/android/build.gradle (hunk 1 above)
Comment (geriux) [reply packages/react-native-aztec/android/build.gradle:14]: I'll do the writing flow tests in https://github.com/wordpress-mobile/gutenberg-mobile/pull/6196#issuecomment-1733404065 to double check it's working as expected.

**Review by geriux — APPROVED** [review]
LGTM! 🚀 Thanks for working on this!

**Pull request merged**